src/cassette/CassettePlayer.cc

```diff
@@ -322,11 +323,20 @@ void CassettePlayer::insertTape(const Filename& filename)
 					filename, filePool,
 					motherBoard.getMSXCliComm());
 			} catch (MSXException& e2) {
-				throw MSXException(
-					"Failed to insert WAV image: \"" +
-					e.getMessage() +
-					"\" and also failed to insert CAS image: \"" +
-					e2.getMessage() + '\"');
+				try {
+					// if that fails use TSX
+					playImage = make_unique<TsxImage>(
+						filename, filePool,
+						motherBoard.getMSXCliComm());
+				} catch (MSXException& e3) {
+					throw MSXException(
+						"Failed to insert WAV image: \"" +
+						e.getMessage() +
+						"\", also failed to insert CAS image: \"" +
+						e2.getMessage() +
+						"\" and also failed to insert TSX image: \"" +
+						e3.getMessage() + '\"');
+				}
```

Comment:
We are nesting too much try-catch statements here. Perhaps we should extract it to other functions:

```c++
void CassettePlayer::insertTape(const Filename& filename)
{
    try { ... }
    catch (MSXException& e) {
        insertCasTape(filename);
    }
}

void CassetePlayer::insertCasTape(const Filename& filename)
{
    try {
        // The code that try to load the CAS tape
    } catch (MSXException& e) {
        insertTsxTape(filename);
    }
}

void CassetePlayer::insertTsxTape(const Filename& filename)
{
    // Your code here
}
```